api/src/services/search-index-service.ts

```diff
@@ -0,0 +1,97 @@
+import { IDBConnection } from '../database/db';
+import {
+  FeaturePropertyRecordWithPropertyTypeName,
+  Geometry,
+  InsertDatetimeSearchableRecord,
+  InsertNumberSearchableRecord,
+  InsertSpatialSearchableRecord,
+  InsertStringSearchableRecord,
+  SearchIndexRepository
+} from '../repositories/search-index-respository';
+import { SubmissionRepository } from '../repositories/submission-repository';
+import { getLogger } from '../utils/logger';
+import { DBService } from './db-service';
+
+const defaultLog = getLogger('services/search-index-service');
+
+export class SearchIndexService extends DBService {
+  searchIndexRepository: SearchIndexRepository;
+
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.searchIndexRepository = new SearchIndexRepository(connection);
+  }
+
+  /**
+   * Creates search indexes for datetime, number, spatial and string properties belonging to
+   * all features found for the given submission.
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<void>}
+   * @memberof SearchIndexService
+   */
+  async indexFeaturesBySubmissionId(submissionId: number): Promise<void> {
+    defaultLog.debug({ label: 'indexFeaturesBySubmissionId' });
+
+    const datetimeRecords: InsertDatetimeSearchableRecord[] = [];
+    const numberRecords: InsertNumberSearchableRecord[] = [];
+    const spatialRecords: InsertSpatialSearchableRecord[] = [];
+    const stringRecords: InsertStringSearchableRecord[] = [];
+
+    const submissionRepository = new SubmissionRepository(this.connection);
+    const features = await submissionRepository.getSubmissionFeaturesBySubmissionId(submissionId);
+
+    const featurePropertyTypeNames: FeaturePropertyRecordWithPropertyTypeName[] =
+      await this.searchIndexRepository.getFeaturePropertiesWithTypeNames();
+    const featurePropertyTypeMap: Record<string, FeaturePropertyRecordWithPropertyTypeName> = Object.fromEntries(
+      featurePropertyTypeNames.map((propertyType) => {
+        const { name } = propertyType;
+        return [name, propertyType];
+      })
+    );
+
+    features.forEach((feature) => {
+      const { submission_feature_id } = feature;
+      Object.entries(feature.data.properties).forEach(([feature_property_name, value]) => {
```

Comment:
Should just be `feature.data` rather than `feature.data.properties`

Reply:
Fixed